File: SpotifyAPI/Models/Entities/Playlist.cs
using Microsoft.EntityFrameworkCore;

namespace Models.Entities;

[PrimaryKey("Id")]
public class Playlist : Entity
{
    public string PreviewId { get; set; } = default!;
    public User Owner { get; set; } = default!;
    public string OwnerId { get; set; } = default!;
    public List<Track> Tracks { get; set; } = new();

    public Playlist(string name, User owner, string previewId = "default_playlist")
    {
        Name = name;
        Owner = owner;
        PreviewId = previewId;
        Tracks = new List<Track>();
    }

    public Playlist(string name, string ownerId, string previewId = "default_playlist")
    {
        Name = name;
        OwnerId = ownerId;
        PreviewId = previewId;
    }

    public Playlist(string id, string name, User owner, string previewId)
    {
        Id = id;
        Name = name;
        Owner = owner;
        PreviewId = previewId;
    }

    public Playlist()
    {
    }
}

File: spotik_mobile/lib/models/entity/playlist_light/playlist.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:spotik_mobile/models/entity/user_light/user.dart';

part 'playlist.freezed.dart';
part 'playlist.g.dart';

@freezed
class Playlist with _$Playlist {
  // ignore: invalid_annotation_target
  @JsonSerializable(explicitToJson: true)
  const factory Playlist (
      {
        required String id,
        required String previewId,
        required String name,
        required User owner,
        required int trackCount,
      }) = _Playlist;

  factory Playlist.fromJson(Map<String, dynamic> json) => _$PlaylistFromJson(json);
}